### ISSUE TITLE
{lubridate} Duration type (`<dtrn>`) incorrectly shows as `<time>` within R markdown editor code chunks

### ISSUE THREAD

**Issue by mstackhouse** [rstudio/rstudio#9772]
### System details

    RStudio Edition : RStudio Open-Source IDE (and RStudio Server Pro)
    RStudio Version : Open-Source version 1.4.1717, RSP Version 1.4.1103-1 (and 
    OS Version      : Windows 10 and Ubuntu 18.04.5 LTS
    R Version       :  R 4.0.2

### Steps to reproduce the problem

Run this code in the console:

``` r
x <- tibble::tribble(
  ~D1, ~D2,
  lubridate::as_date("2020-01-01"), lubridate::as_date("2019-06-19")
)

dplyr::mutate(x,
              D3 = D1 - D2
)
#> # A tibble: 1 x 3
#>   D1         D2         D3      
#>   <date>     <date>     <drtn>  
#> 1 2020-01-01 2019-06-19 196 days
```

<sup>Created on 2021-08-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Create a markdown document, place this in a code chunk, and then run the code chunk. 

### Describe the problem in detail

The code run in the console presents the correct results, as can be seen in the reprex. The variable `D3` should be of type `<dtrn>`. When the code is run the R markdown chunk, the type of D3 shows as `<time>`:

![image](https://user-images.githubusercontent.com/31387567/130655987-c6387210-d090-4eff-93b9-3ad09e16a041.png)

### Describe the behavior you expected

The HTML table produced by the code chunk should display the type of D3 as `<time>`. Please note that this issues does NOT occur when the R Markdown document is rendered. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by mstackhouse** [comment]
Hey @ronblum it looks like this went stale - but it was acknowledged as a bug so just curious if this should be kept open?

**Comment by ronblum** [comment]
@mstackhouse Yes, the issues automatically close if there's no activity, and making a note—you just did—unstales it so that it doesn't close. Unfortunately, I can provide any information about fixing this. We have lots of issues that we prioritize in a few ways, but one way in which we prioritize issues—bugs and enhancements—is by the number of upvotes in the original post, so if you can recruit more upvotes, this is more likely to be addressed.

Note for triage: Can we consider this for Spotted Wakerobin or Elsbeth Geranium, since wrong information is being produced? It's not a regression, though—this has been around since at lease 1.3.1093.
